network/tls_authentication.go

```diff
@@ -1,10 +1,73 @@
 package network
 
 import (
+	"crypto/sha512"
 	"crypto/x509"
+	"encoding/hex"
 	"errors"
+	"github.com/cossacklabs/acra/keystore"
 )
 
+// CertificateIdentifierExtractor interface for implementations which should return identifier used for client's identification
+type CertificateIdentifierExtractor interface {
+	GetCertificateIdentifier(certificate *x509.Certificate) ([]byte, error)
+}
+
+// CommonNameExtractor implementation for CertificateIdentifierExtractor interface, which return CommonName as client's identifier
+type CommonNameExtractor struct{}
```

Comment:
I don't want to be that nitpicking jerk with 50 spelling comments for each PR, but the `certificate.Subject` this extractor returns is the "distinguished name". The "common name" is a part of a distinguished name. I believe this extractor should be renamed to avoid confusion. (Or maybe you intended to extract *just* the common name?)

Reply:
yes, agree. I will change. main reason was to use "common name" because saw that this name usually used in discussing tls fields and identification